app/Http/Controllers/VolunteerController.php

```diff
@@ -60,5 +61,65 @@ public static function show($id)
         return view('volunteer.show', compact('volunteer'));
     }
 
+    /**
+     *  Volunteer set a challenge to himself
+     */
+    public function createChallenge()
+    {
+            return view('volunteer.challenge');
+
+    }
+
+
+    /**
+     *  Store the challenge in the database
+     */
+    public function storeChallenge(Request $request)
+    {
+        $this->validate($request ,
+            ['events' => 'required|numeric|min:1' ,
+             'deadline' => 'required|date_format:Y-d-m'
+            ]);
+        $user_id = Auth::user()->id;
```

Comment:
the challenges are designed for a whole year (similar to that implement in goodreads. That is, the volunteer is promoted to set a number of events to attend in the current year